### DISCUSSION TITLE
Changing cephcluster config from /dev/sda to SCSI ID, is it supported ?

### DISCUSSION THREAD

**Discussion by ssjafri** [rook/rook#14662] [Q&A]
I had a cephcluster on openshift running & had osds defined by disk name : /sda (osd0 - osd5)

we were facing some issues in which there disks were getting randomly name changed after boot. So, checking into docs config for specific nodes & devices, 
https://rook.io/docs/rook/latest-release/CRDs/Cluster/host-cluster/#specific-nodes-and-devices

i have made changes in the cluster OSD definition from name to scsi-id

Now while the cluster has made new OSDs (OSD6 - OSD11) out of the same & its running fine, i am seeing the old OSD deployment & pods still getting scaled up.

i have cleaned OSD definition from ceph c
|eph osd down osd.<numer OSD>
ceph osd out osd.<numer OSD>
ceph osd crush remove osd.<numer OSD>
ceph auth del osd.<number OSD>
ceph osd rm osd.<numer OSD>

also cleaned from disk

ssh core@<node where OSD was> "sudo rm -vrf /var/lib/rook/"

My question is: 
1. Is it supported to change on a running cluster from dev name to scsi-id for same backend devices ?
2. if above is yes then how do i permanently disable old OSDs deployment

Thanks in advance

**Comment by travisn** [comment]
What are your settings in the cephcluster CR for the deviceFilter and related storage settings? It sounds like the disks are still being identified with the old names. You likely need a `deviceFilter` for new names instead.

**Comment by szczepl** [comment]
This is how we defined this:

    storage:
      nodes:
      - devices:
        - name: /dev/disk/by-id/scsi-362cea7f0d26673002e1785d837683024
        name: xxx-ops01
      - devices:
        - name: /dev/disk/by-id/scsi-36f4ee0806b6014002e1769aabd121b28
        name: xxx-ops02
      - devices:
        - name: /dev/disk/by-id/scsi-36f4ee0806b603d002e1769b8c178fb9f
        name: xxx-ops03
      - devices:
        - name: /dev/disk/by-id/scsi-362cea7f0d25248002e1769c202f65885
        name: xxx-ops06
      - devices:
        - name: /dev/disk/by-id/scsi-36f4ee080619a2a002e1769d3d106ed7a
        name: xxx-ops09
      - devices:
        - name: /dev/disk/by-id/scsi-36f4ee080619a2b002e1769e5beb3441b
        name: xxx-ops10
      - devices:
        - name: /dev/disk/by-id/scsi-36f4ee080619a38002e1769f2bd9925fc
        name: xxx-ops11
      - devices:
        - name: /dev/disk/by-id/scsi-36f4ee080619a29002e178a8574707168
        name: xxx-ops12
      useAllDevices: false
      useAllNodes: false

However, after every pipeline run we are finding OSDs in CrashLoopBack and rook is still trying to reach sdX devices, instead of these above.

**Reply by travisn** [reply]
The OSD prepare pod log for each host should explain how it is finding the original OSD, and probably has a message about the new device path already containing an OSD since it's the same device. You likely need to go through and [purge each OSD](https://rook.io/docs/rook/latest/Storage-Configuration/Advanced/ceph-osd-mgmt/#remove-an-osd) one-by-one (and wait for data backfill for PGs to be healthy) and allow it to be provisioned with the new name.
